packages/@lwc/integration-karma/test/rendering/callback-invocation-order/index.spec.js

```diff
@@ -228,3 +229,54 @@ it('should invoke callbacks on the right order (issue #1199 and #1198)', () => {
         );
     });
 });
+
+it('should invoke callbacks on the right order when multiple templates are used with lwc:if', () => {
+    const elm = createElement('x-multi-template-conditionals', { is: MultiTemplateConditionals });
+    elm.show = true;
+    document.body.appendChild(elm);
+
+    // initial load is x-shadow-parent
+    expect(window.timingBuffer).toEqual([
+        'leaf:T1-1:connectedCallback',
+        'leaf:T1-2:connectedCallback',
+        'leaf:T1-3:connectedCallback',
+        'leaf:T1-4:connectedCallback',
+        'leaf:T1-5:connectedCallback',
+        'leaf:T1-6:connectedCallback',
+    ]);
+
+    resetTimingBuffer();
+    elm.next();
+
+    return Promise.resolve()
+        .then(() => {
+            // disconnect x-shadow-parent +
+            // connect x-shadow-container with 2 parents, 'a' and 'b'
+            expect(window.timingBuffer).toEqual([
```

Comment:
I checked the ordering here in both synthetic and native shadow against if:true and lwc:if:
```
if:true - synthetic - 654321
lwc:if - synthetic - 654321
if:true - native - 123456
lwc:if - native - 654321
```
In synthetic, both if:true and lwc:if are managed by `runChildNodesDisconnectedCallback`, which uses reverse ordering.
In native, if:true is managed by `resetComponentRoot`, which uses forward ordering.

`lwc:if` in native cannot match both the old `if:true` behavior (forward) and the `runChildNodesDisconnectedCallback` behavior (reverse). From https://github.com/salesforce/lwc/issues/2713 it sounds like forward is supposed to be the "correct" behavior, and that's the behavior when conditionals aren't used at all.

IMO since `if:true` already has different behaviors between synthetic and native, doing the same for `lwc:if` shouldn't be introducing any confusion that didn't exist before. Keep it consistent (reverse) for synthetic, and correct (forward) for native:
```
if:true - synthetic - 654321
lwc:if - synthetic - 654321
if:true - native - 123456
lwc:if - native - 123456
```

Do you see any problems with this approach?

Reply:
Note that it would not be too difficult to update lwc:if to use forward ordering; it already goes through `resetComponentRoot` which already does so.

Reply:
Conclusion from Dev Sync:
```
if:true - synthetic - 654321
lwc:if - synthetic - 654321
if:true - native - 123456
lwc:if - native - 123456
```